src/App.js

```diff
@@ -25,17 +25,21 @@ const OkendoStarRating = (props) => {
   const widgetContainer = useRef(null);
 
   const initialiseReviewsWidget = () => {
-    window.okeWidgetApi.initWidget(widgetContainer.current);
+    // if window.okeWidgetApi.initWidget is available
+    if (typeof window.okeWidgetApi?.initWidget === 'function' && widgetContainer.current !== null) {
+      window.okeWidgetApi.initWidget(widgetContainer.current);
+    }
   }
 
   useEffect(() => {
     if (window.okeWidgetApi) {
-      initialiseReviewsWidget();
+        initialiseReviewsWidget(); 
     }
     else {
       window.addEventListener('oke-script-loaded', initialiseReviewsWidget);
     }
-  }, []);
+
+  }, [props.productId]); // only re-run the effect if props.productId changes
```

Comment:
(Ah yep, that's useful, I've included it in my PR :wink: Thank you!)

Reply:
I have also one suggestion we should also unregister the event from `windows` object  when component is _`unmounted from DOM_`

Reply:
Ah yes, good idea. Would you like to fix the conflicts in your PR and add this, or are you OK for me to do it?

Reply:
I am fixing the issue in my branch and hope it will be merged